cdc/sink/codec/avro.go

```diff
@@ -0,0 +1,331 @@
+		data:       bin,
+		registryID: registryID,
+	}, nil
+}
+
+type avroSchemaTop struct {
+	Tp     string            `json:"type"`
+	Name   string            `json:"name"`
+	Fields []avroSchemaField `json:"fields"`
+}
+
+type avroSchemaField struct {
+	Name         string      `json:"name"`
+	Tp           []string    `json:"type"`
+	DefaultValue interface{} `json:"default"`
+}
+
+func columnInfoToAvroSchema(name string, columnInfo []*timodel.ColumnInfo) (string, error) {
+	top := avroSchemaTop{
+		Tp:     "record",
+		Name:   name,
+		Fields: nil,
+	}
+
+	for _, col := range columnInfo {
+		field := avroSchemaField{
+			Name:         col.Name.String(),
+			Tp:           []string{"null", getAvroDataTypeNameMysql(col.Tp)},
+			DefaultValue: nil,
+		}
+		top.Fields = append(top.Fields, field)
+	}
+
+	str, err := json.Marshal(&top)
+	if err != nil {
+		return "", errors.Annotate(err, "columnInfoToAvroSchema: failed to generate json")
+	}
+	return string(str), nil
+}
+
+func rowToAvroNativeData(cols map[string]*model.Column) (interface{}, error) {
+	ret := make(map[string]interface{}, len(cols))
+	for key, col := range cols {
+		data, str, err := columnToAvroNativeData(col)
+		if err != nil {
+			return nil, err
+		}
+
+		union := make(map[string]interface{}, 1)
+		union[str] = data
+		ret[key] = union
+	}
+	return ret, nil
+}
+
+func getAvroDataTypeName(v interface{}) string {
+	switch v.(type) {
+	case bool:
+		return "boolean"
+	case []byte:
+		return "bytes"
+	case float64:
+		return "double"
+	case float32:
+		return "float"
+	case int64, uint64:
+		return "long"
+	case int, int32, uint32:
+		return "int"
+	case nil:
+		return "null"
+	case string:
+		return "string"
+	case time.Duration:
+		return "long.time-millis"
+	case time.Time:
+		return "long.timestamp-millis"
+	default:
+		log.Warn("getAvroDataTypeName: unknown type")
+		return "errorType"
+	}
+}
+
+func getAvroDataTypeNameMysql(tp byte) string {
+	switch tp {
+	case mysql.TypeFloat:
+		return "float"
+	case mysql.TypeDouble:
+		return "double"
+	case mysql.TypeVarchar, mysql.TypeString, mysql.TypeVarString:
+		return "string"
+	case mysql.TypeDate, mysql.TypeDatetime, mysql.TypeTimestamp:
+		return "long.timestamp-millis"
+	case mysql.TypeDuration: //duration should read fsp from column meta data
+		return "long.time-millis"
+	case mysql.TypeEnum:
+		return "long"
+	case mysql.TypeSet:
+		return "long"
+	case mysql.TypeBit:
+		return "long"
+	case mysql.TypeNewDecimal, mysql.TypeDecimal:
+		return "string"
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24:
+		return "int"
+	case mysql.TypeLong, mysql.TypeLonglong:
+		return "long"
+	case mysql.TypeNull:
+		return "null"
+	case mysql.TypeJSON:
+		return "string"
+	case mysql.TypeTinyBlob, mysql.TypeMediumBlob, mysql.TypeLongBlob, mysql.TypeBlob:
+		return "bytes"
+	default:
+		return "error"
+	}
+}
+
+func columnToAvroNativeData(col *model.Column) (interface{}, string, error) {
+	if v, e := col.Value.(int); e {
```

Comment:
```suggestion
	if v, ok := col.Value.(int); ok {
```
`ok` is used more often in type assertion in golang.